src/build/helpers.ts

```diff
@@ -114,6 +114,25 @@ export function exposesTo(o: { exposed?: string }, target: string[]): boolean {
   return o.exposed.split(" ").some((e) => target.includes(e));
 }
 
+export function clone<T>(obj: T): T {
+  if (typeof obj !== "object" || !obj) {
+    return obj;
+  }
+  if (Array.isArray(obj)) {
+    return obj.map(clone) as any as T;
+  }
+  if (obj instanceof Map) {
+    return new Map([...obj.entries()].map(clone)) as any as T;
+  }
+  const result = {} as T;
+  for (const k in obj) {
+    if (Object.getOwnPropertyDescriptor(obj, k)) {
```

Comment:
`Object.hasOwn` might be better.

```suggestion
    if (Object.hasOwn(obj, k)) {
```